jax/interpreters/pxla.py

```diff
@@ -394,13 +407,19 @@ def __getitem__(self, idx):
 def xla_pmap_impl(fun, *args, **params):
   axis_name = params.pop('axis_name')
   axis_size = params.pop('axis_size')
+  devices = params.pop('devices')
   assert not params
+
+  if devices is not None and len(devices) == 0:
```

Comment:
This particular check is super cheap, but I recently got the idea from Peter that we could do all our error checking under the cached functions (i.e. under `parallel_callable` here), since that means we only do the error checking once (at "compile time") and not on each dispatch.